lib/SILOptimizer/Transforms/SpeculativeDevirtualizer.cpp

```diff
@@ -257,18 +257,8 @@ static bool isDefaultCaseKnown(ClassHierarchyAnalysis *CHA,
     return false;
 
   // Only consider 'private' members, unless we are in whole-module compilation.
-  switch (CD->getEffectiveAccess()) {
-  case AccessLevel::Open:
+  if (isDeclVisibleExternally(CD, &AI.getModule()))
     return false;
-  case AccessLevel::Public:
```

Comment:
The comment says we should only consider private members, but it treats public members the same as internal members, which makes me think I'm possible misunderstanding this.

Reply:
It's probably just an oversight.

Reply:
Ok. I think something in this later change broke a SILOptimizer test (global_property_opt.sil), but I wasn't yet able to figure out what it meant.

Here's a gist with the difference: https://gist.github.com/dtweston/1ce13a14a0cadbfe2088aa647ee7a891

Reply:
This test breakage has gone away, so I think everything's fine now.